### PR TITLE
feat(rpc): impl Filecoin.NetProtectAdd and NetProtectRemove

### LINKED ISSUE #4753: Implement `Filecoin.NetProtectRemove`

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

Changes introduced in this pull request:

- impl Filecoin.NetProtectAdd and NetProtectRemove
- removes `#[allow(dead_code)]` from https://github.com/ChainSafe/forest/pull/4742/files#diff-4c7830ac5276b0c3eac19556e3ef6814eff768b6af5d52682a1bc772af1d870cR352
- api compare tests

## Reference issue to close (if applicable)

Closes #4753

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by LesnyRumcajs — COMMENTED** [review]
Is it feasible to add tests here (even if it's just calling the method and checking if it's not panicking)?

**Comment by hanabi1224** [comment]
> Is it feasible to add tests here (even if it's just calling the method and checking if it's not panicking)?

@LesnyRumcajs The current test infrastructure does not support invoking RPC methods with JWT token. I'd like to fix that in a separate PR and add tests for methods like `Filecoin.AuthNew` as well. What do you think?

**Comment by LesnyRumcajs** [comment]
> > Is it feasible to add tests here (even if it's just calling the method and checking if it's not panicking)?
> 
> @LesnyRumcajs The current test infrastructure does not support invoking RPC methods with JWT token. I'd like to fix that in a separate PR and add tests for methods like `Filecoin.AuthNew` as well. What do you think?

It _does_ support it, I implemented it myself. :) https://github.com/ChainSafe/forest/blob/589570c487e0f28e7c57f7858aec7fa0d1fdf1f7/scripts/tests/api_compare/docker-compose.yml#L209-L210

Or am I missing something?

**Comment by hanabi1224** [comment]
> It does support it, I implemented it myself. :)

@LesnyRumcajs Nice, tests added. I will add tests for methods like `AuthNew` as well in another PR.

File: src/libp2p/service.rs [hunk 1]
```diff
@@ -500,12 +500,16 @@ async fn handle_network_message(
...
+                        peer_manager.protect_peer(peer_id);
+                    }
                     tx.send_or_warn(());
                 }
-                NetRPCMethods::UnprotectPeer(tx, peer_id) => {
-                    peer_manager.unprotect_peer(&peer_id);
+                NetRPCMethods::UnprotectPeer(tx, peer_ids) => {
+                    for peer_id in peer_ids {
+                        peer_manager.unprotect_peer(&peer_id);
+                    }
```
Comment (sudo-shashank) [inline src/libp2p/service.rs:512]: nit:
```suggestion
                    peer_ids.iter().for_each(|peer_id| {
    peer_manager.unprotect_peer(peer_id);
});
```

File: src/libp2p/service.rs [hunk 2]
```diff
@@ -500,12 +500,16 @@ async fn handle_network_message(
                     let peer_addresses = swarm.behaviour().peer_addresses();
                     response_channel.send_or_warn(peer_addresses);
                 }
-                NetRPCMethods::ProtectPeer(tx, peer_id) => {
-                    peer_manager.protect_peer(peer_id);
+                NetRPCMethods::ProtectPeer(tx, peer_ids) => {
+                    for peer_id in peer_ids {
+                        peer_manager.protect_peer(peer_id);
+                    }
```
Comment (sudo-shashank) [inline src/libp2p/service.rs:506]: nit:
```suggestion
                    peer_ids.iter().for_each(|peer_id| {
    peer_manager.protect_peer(peer_id);
});
```

**Review by LesnyRumcajs — APPROVED** [review]

**Review by sudo-shashank — APPROVED** [review]

File: src/libp2p/service.rs (hunk 2 above)
Comment (hanabi1224) [reply src/libp2p/service.rs]: Fixed.

File: src/libp2p/service.rs (hunk 1 above)
Comment (hanabi1224) [reply src/libp2p/service.rs]: Fixed.

**Review by ruseinov — APPROVED** [review]

**Pull request merged**